File: crates/server/src/providers/email.rs
//! Mock Email provider implementation (Feature 008 - US1)

use crate::config::ApiConfig;
use crate::providers::common::pick_outcome_for_provider;
use crate::providers::mock::Outcome;
use crate::providers::registry::{DispatchResult, OutboundMessage, Provider};

#[derive(Debug, Clone)]
pub struct EmailMockProvider;

impl Default for EmailMockProvider {
    fn default() -> Self {
        Self
    }
}

impl EmailMockProvider {
    pub fn new() -> Self {
        Self
    }
}

impl Provider for EmailMockProvider {
    fn name(&self) -> &str {
        "email"
    }
    fn dispatch(&self, _msg: &OutboundMessage, cfg: &ApiConfig) -> DispatchResult {
        let (outcome, _roll): (Outcome, u32) = pick_outcome_for_provider(self.name(), cfg);
        DispatchResult {
            provider_name: self.name().to_string(),
            outcome,
        }
    }
}

File: go/internal/providers/mock/email.go
package mock

import (
    "context"
    "time"

    "github.com/sj4nes/messaging-service/go/internal/providers"
)

type EmailProvider struct{}

func NewEmailProvider() *EmailProvider { return &EmailProvider{} }

func (e *EmailProvider) Name() string { return "email" }

func (e *EmailProvider) Dispatch(ctx context.Context, msg providers.OutboundMessage) providers.DispatchResult {
    select {
    case <-ctx.Done():
        return providers.DispatchResult{ProviderName: e.Name(), Outcome: providers.OutcomeTimeout}
    case <-time.After(10 * time.Millisecond):
    }
    return providers.DispatchResult{ProviderName: e.Name(), Outcome: providers.OutcomeSuccess}
}

var _ providers.Provider = (*EmailProvider)(nil)
